### PR TITLE
[state]: state api refactor.

### PR DESCRIPTION
**Pull request by tanghaodong25**
1. refactor store api.
2. add store development document [README](https://github.com/TuGraph-family/tugraph-analytics/blob/671d269cadfa695687a2634136e9ccd3ee600e55/geaflow/geaflow-plugins/geaflow-store/README.md).
3. delete seldom used features.

### CODE REVIEW

**Review by cfsfine — APPROVED** [review]
LGTM.

File: geaflow/geaflow-plugins/geaflow-store/README.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+- `close()`: Close storage connection
+
+### IStatefulStore
+
+State management storage interface, inheriting from IBaseStore, providing stateful management
+operations:
+
+- `archive()`: Archive current state
+- `recovery()`: Recover state from specified position
+- `recoveryLatest()`: Recover state from specified position
```
Comment (Loognqiang) [inline geaflow/geaflow-plugins/geaflow-store/README.md]: The descriptions for the recoveryLatest and recovery interfaces should be distinguished from each other.

Comment (tanghaodong25) [reply geaflow/geaflow-plugins/geaflow-store/README.md]: @Loognqiang issue has been resolved.

File: geaflow/geaflow-plugins/geaflow-store/geaflow-store-api/src/main/java/com/antgroup/geaflow/store/iterator/KeysIterator.java [hunk 2]
```diff
@@ -30,15 +30,13 @@ public class KeysIterator<K, VV, EV, R> implements CloseableIterator<R> {
...
-        IStatePushDown pushdown) {
+    public KeysIterator(List<K> keys, BiFunction<K, IStatePushDown, R> fetchFun,
+                        IStatePushDown pushdown) {
         this.fetchFun = fetchFun;
         this.iterator = keys.iterator();
         if (pushdown.getFilters() != null) {
             StatePushDown simpleKeyPushDown = StatePushDown.of()
-                .withEdgeLimit(pushdown.getEdgeLimit())
-                .withOrderFields(pushdown.getOrderFields());
+                .withEdgeLimit(pushdown.getEdgeLimit()).withOrderFields(pushdown.getOrderFields());
```
Comment (Loognqiang) [inline geaflow/geaflow-plugins/geaflow-store/geaflow-store-api/src/main/java/com/antgroup/geaflow/store/iterator/KeysIterator.java]: The change has resulted in an unreasonable format

File: geaflow/geaflow-plugins/geaflow-store/geaflow-store-cstore/src/main/java/com/antgroup/geaflow/store/cstore/StaticGraphCStore.java [hunk 3]
```diff
@@ -136,13 +137,17 @@ private void rewriteConfig() {
...
+                String endpoint = Configuration.getString(FileConfigKeys.OSS_ENDPOINT,
+                    persistConfig);
                 this.config.put(CStoreConfigKeys.CSTORE_OSS_ENDPOINT, endpoint);
-                String accessKeyId = Configuration.getString(FileConfigKeys.OSS_ACCESS_ID, persistConfig);
+                String accessKeyId = Configuration.getString(FileConfigKeys.OSS_ACCESS_ID,
+                    persistConfig);
                 this.config.put(CStoreConfigKeys.CSTORE_OSS_AK, accessKeyId);
-                String accessKeySecret = Configuration.getString(FileConfigKeys.OSS_SECRET_KEY, persistConfig);
+                String accessKeySecret = Configuration.getString(FileConfigKeys.OSS_SECRET_KEY,
+                    persistConfig);
```
Comment (Loognqiang) [inline geaflow/geaflow-plugins/geaflow-store/geaflow-store-cstore/src/main/java/com/antgroup/geaflow/store/cstore/StaticGraphCStore.java]: Unnecessary format

File: geaflow/geaflow-plugins/geaflow-store/geaflow-store-jdbc/src/main/java/com/antgroup/geaflow/store/jdbc/BaseJdbcStore.java [hunk 4]
```diff
@@ -72,23 +72,22 @@ private synchronized void initConnectionPool(Configuration config) {
...
-                                   Object[] values) throws SQLException {
+    protected boolean insert(String key, String[] columns, Object[] values) throws SQLException {
         if (columns.length != values.length) {
             throw new GeaflowRuntimeException("columns' size does not match values'");
         }
 
-        String sql = String.format(insertFormat,
-            this.tableName, this.pk, Joiner.on(',').join(columns),
+        String sql = String.format(insertFormat, this.tableName, this.pk,
+            Joiner.on(',').join(columns),
```
Comment (Loognqiang) [inline geaflow/geaflow-plugins/geaflow-store/geaflow-store-jdbc/src/main/java/com/antgroup/geaflow/store/jdbc/BaseJdbcStore.java]: `,` should be defined constant value

File: geaflow/geaflow-plugins/geaflow-store/geaflow-store-jdbc/src/test/java/com/antgroup/geaflow/store/jdbc/JdbcStoreBuilderTest.java [hunk 5]
```diff
@@ -34,11 +34,8 @@ public class JdbcStoreBuilderTest {
 
     private void prepareSqlite() throws Exception {
         String createSqliteTable = "CREATE TABLE IF NOT EXISTS `store_test`(\n"
-            + "   `id` bigint UNSIGNED AUTO_INCREMENT,\n"
-            + "   `pk` VARCHAR(256) NOT NULL,\n"
-            + "   `value` LONGBLOB,\n"
-            + "   PRIMARY KEY ( `id`)\n"
-            + ")";
+            + "   `id` bigint UNSIGNED AUTO_INCREMENT,\n" + "   `pk` VARCHAR(256) NOT NULL,\n"
```
Comment (Loognqiang) [inline geaflow/geaflow-plugins/geaflow-store/geaflow-store-jdbc/src/test/java/com/antgroup/geaflow/store/jdbc/JdbcStoreBuilderTest.java]: as above

File: geaflow/geaflow-plugins/geaflow-store/geaflow-store-memory/src/main/java/com/antgroup/geaflow/store/memory/BaseStaticGraphMemoryStore.java [hunk 6]
```diff
@@ -96,35 +98,41 @@ public OneDegreeGraph<K, VV, EV> getOneDegreeGraph(K sid, IStatePushDown pushdow
...
     public CloseableIterator<OneDegreeGraph<K, VV, EV>> getOneDegreeGraphIterator(
         IStatePushDown pushdown) {
-        return new KeysIterator<>(Lists.newArrayList(getKeyIterator()), this::getOneDegreeGraph, pushdown);
+        return new KeysIterator<>(Lists.newArrayList(getKeyIterator()), this::getOneDegreeGraph,
+            pushdown);
     }
 
     @Override
-    public CloseableIterator<OneDegreeGraph<K, VV, EV>> getOneDegreeGraphIterator(List<K> keys, IStatePushDown pushdown) {
+    public CloseableIterator<OneDegreeGraph<K, VV, EV>> getOneDegreeGraphIterator(List<K> keys,
```
Comment (Loognqiang) [inline geaflow/geaflow-plugins/geaflow-store/geaflow-store-memory/src/main/java/com/antgroup/geaflow/store/memory/BaseStaticGraphMemoryStore.java]: There exists a large amount of unnecessary formatting

File: geaflow/geaflow-plugins/geaflow-store/geaflow-store-memory/src/main/java/com/antgroup/geaflow/store/memory/MemoryDynamicGraphStore.java [hunk 7]
```diff
@@ -47,13 +47,13 @@
 import java.util.function.BiFunction;
 import java.util.stream.Collectors;
 
-public class GraphMemoryMultiVersionedStore<K, VV, EV> extends BaseGraphStore implements
-    IGraphMultiVersionedStore<K, VV, EV> {
+public class MemoryDynamicGraphStore<K, VV, EV> extends BaseGraphStore implements
```
Comment (Loognqiang) [inline geaflow/geaflow-plugins/geaflow-store/geaflow-store-memory/src/main/java/com/antgroup/geaflow/store/memory/MemoryDynamicGraphStore.java]: MemoryDynamicGraphStore -> DynamicGraphMemoryStore

File: geaflow/geaflow-plugins/geaflow-store/geaflow-store-rocksdb/src/main/java/com/antgroup/geaflow/store/rocksdb/DynamicRocksdbGraphStoreBase.java [hunk 8]
```diff
@@ -35,8 +35,8 @@
 import java.util.List;
 import java.util.Map;
 
-public class GraphMultiVersionedRocksdbStore<K, VV, EV> extends BaseRocksdbGraphStore
-    implements IGraphMultiVersionedStore<K, VV, EV> {
+public class DynamicRocksdbGraphStoreBase<K, VV, EV> extends BaseRocksdbGraphStore
```
Comment (Loognqiang) [inline geaflow/geaflow-plugins/geaflow-store/geaflow-store-rocksdb/src/main/java/com/antgroup/geaflow/store/rocksdb/DynamicRocksdbGraphStoreBase.java]: Naming conventions should remain consistent, DynamicRocksdbGraphStoreBase -> DynamicGraphRocksdbStoreBase

File: geaflow/geaflow-plugins/geaflow-store/geaflow-store-api/src/main/java/com/antgroup/geaflow/store/iterator/KeysIterator.java (hunk 2 above)
Comment (tanghaodong25) [reply geaflow/geaflow-plugins/geaflow-store/geaflow-store-api/src/main/java/com/antgroup/geaflow/store/iterator/KeysIterator.java]: issue has been resolved.

File: geaflow/geaflow-plugins/geaflow-store/geaflow-store-cstore/src/main/java/com/antgroup/geaflow/store/cstore/StaticGraphCStore.java (hunk 3 above)
Comment (tanghaodong25) [reply geaflow/geaflow-plugins/geaflow-store/geaflow-store-cstore/src/main/java/com/antgroup/geaflow/store/cstore/StaticGraphCStore.java]: issue has been resolved.

File: geaflow/geaflow-plugins/geaflow-store/geaflow-store-jdbc/src/main/java/com/antgroup/geaflow/store/jdbc/BaseJdbcStore.java (hunk 4 above)
Comment (tanghaodong25) [reply geaflow/geaflow-plugins/geaflow-store/geaflow-store-jdbc/src/main/java/com/antgroup/geaflow/store/jdbc/BaseJdbcStore.java]: issue has been resolved.

File: geaflow/geaflow-plugins/geaflow-store/geaflow-store-jdbc/src/test/java/com/antgroup/geaflow/store/jdbc/JdbcStoreBuilderTest.java (hunk 5 above)
Comment (tanghaodong25) [reply geaflow/geaflow-plugins/geaflow-store/geaflow-store-jdbc/src/test/java/com/antgroup/geaflow/store/jdbc/JdbcStoreBuilderTest.java]: issue has been resolved.

File: geaflow/geaflow-plugins/geaflow-store/geaflow-store-memory/src/main/java/com/antgroup/geaflow/store/memory/BaseStaticGraphMemoryStore.java (hunk 6 above)
Comment (tanghaodong25) [reply geaflow/geaflow-plugins/geaflow-store/geaflow-store-memory/src/main/java/com/antgroup/geaflow/store/memory/BaseStaticGraphMemoryStore.java]: issue has been resolved.

File: geaflow/geaflow-plugins/geaflow-store/geaflow-store-memory/src/main/java/com/antgroup/geaflow/store/memory/MemoryDynamicGraphStore.java (hunk 7 above)
Comment (tanghaodong25) [reply geaflow/geaflow-plugins/geaflow-store/geaflow-store-memory/src/main/java/com/antgroup/geaflow/store/memory/MemoryDynamicGraphStore.java]: issue has been resolved.

File: geaflow/geaflow-plugins/geaflow-store/geaflow-store-rocksdb/src/main/java/com/antgroup/geaflow/store/rocksdb/DynamicRocksdbGraphStoreBase.java (hunk 8 above)
Comment (tanghaodong25) [reply geaflow/geaflow-plugins/geaflow-store/geaflow-store-rocksdb/src/main/java/com/antgroup/geaflow/store/rocksdb/DynamicRocksdbGraphStoreBase.java]: issue has been resolved.

**Review by Loognqiang — APPROVED** [review]
LGTM

**Review by dy247846795 — APPROVED** [review]
LGTM
